lib/elixir/lib/application.ex

```diff
@@ -523,10 +523,32 @@ defmodule Application do
     :application.set_env(app, key, value, opts)
   end
 
+  @doc """
+  Puts the environment for multiple apps at the same time.
+
+  It will fail if any of the applications appear more than once
```

Comment:
Opinion: change to `It _might_ fail` as it is an undefined behaviour right now and only happens on the newest OTP as to avoid anyone relying on this behaviour